### ISSUE TITLE
Bootstrap task failure yum postgresql11

### ISSUE THREAD

**Issue by mminglis** [apisnetworks/apnscp#15]
Hey.
Testing bootstrapping, getting yum failures. Relevant message:

``` 
TASK [packages/install : Install packages] ****************************************************************************************************************************
FAILED - RETRYING: Install packages (3 retries left).
FAILED - RETRYING: Install packages (2 retries left).
FAILED - RETRYING: Install packages (1 retries left).
fatal: [localhost]: FAILED! => changed=false 
  attempts: 3
  msg: |-
    warning: /var/cache/yum/x86_64/7/pgdg11/packages/postgresql11-libs-11.2-2PGDG.rhel7.x86_64.rpm: Header V4 DSA/SHA1 Signature, key ID 442df0f8: NOKEY
  
  
    GPG key retrieval failed: [Errno 14] curl#37 - "Couldn't open file /etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG-11"
  rc: 1
```

And in that folder:

```
ls /etc/pki/rpm-gpg/
RPM-GPG-KEY-apnscp-7  RPM-GPG-KEY-CentOS-7  RPM-GPG-KEY-CentOS-Debug-7  RPM-GPG-KEY-CentOS-Testing-7  RPM-GPG-KEY-CLOUDFLARE-2  RPM-GPG-KEY-EPEL-7  RPM-GPG-KEY-PGDG
```

**Comment by msaladna** [comment]
PostgreSQL 11 repo is unstable. `pgdg-centos11-11-2.noarch.rpm` for example was missing as of a couple days ago. RPM key has been renamed, dropping the major version from it. Here's a quick fix to get you back on your feet with 11 without starting over:

```bash
ln -s /etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG{,-11}
cd /usr/local/apnscp/resources/playbooks
ansible-playbook bootstrap.yml
```
Anecdotally, running 11 on a new hosting platform, no issues so far. Appears to be limited to just the Yum repository and limited to the last 48 hours.

*Congrats on your second bug filing* 🥇

**Comment by mminglis** [comment]
That works, cheers.

**Comment by mminglis** [comment]
Now running into an issue building php, output below:

```
TASK [php/build-from-source : Compile PHP 7.2.17]
************************************************************************************************
fatal: [localhost]: FAILED! => changed=true 
  cmd: '/usr/local/apnscp/resources/playbooks/roles/php/build-from-source/files/php.config '
  delta: '0:00:20.572933'
  end: '2019-04-17 04:18:03.724577'
  msg: non-zero return code
  rc: 1
  start: '2019-04-17 04:17:43.151644'
  stderr: |-
    configure: WARNING: unrecognized options: --with-exec-dir
    configure: error: Cannot find libpq-fe.h. Please specify correct PostgreSQL installation path
  stderr_lines:
  - 'configure: WARNING: unrecognized options: --with-exec-dir'
  - 'configure: error: Cannot find libpq-fe.h. Please specify correct PostgreSQL installation path'
```

**Comment by msaladna** [comment]
What's `rpm -qa | grep postgres` report? Should have,

```
postgresql11-devel-11.2-2PGDG.rhel7.x86_64
postgresql11-11.2-2PGDG.rhel7.x86_64
postgresql11-contrib-11.2-2PGDG.rhel7.x86_64
postgresql11-libs-11.2-2PGDG.rhel7.x86_64
postgresql11-server-11.2-2PGDG.rhel7.x86_64
```

And does `/usr/pgsql-11` exist?

On the RPM issue, opened a bug with PostgreSQL team ([#15764](https://www.postgresql.org/message-id/15764-beb1262aa7bb3e41%40postgresql.org))

**Comment by mminglis** [comment]
``` 
> rpm -qa | grep postgres
postgresql10-contrib-10.7-2PGDG.rhel7.x86_64
postgresql10-devel-10.7-2PGDG.rhel7.x86_64
postgresql10-libs-10.7-2PGDG.rhel7.x86_64
postgresql10-server-10.7-2PGDG.rhel7.x86_64
postgresql10-10.7-2PGDG.rhel7.x86_64
postgresql11-libs-11.2-2PGDG.rhel7.x86_64

> ls /usr
bin  etc  games  geos37  include  lib  lib64  libexec  local  pgsql-10  pgsql-11  proj49  sbin  share  src  tmp

> ls /usr/pgsql-11
lib  share

```

**Comment by msaladna** [comment]
Which PostgreSQL release did you opt for, 10 or 11? `grep pgsql_version /root/apnscp-vars.yml`

If 11, then I'm curious how 10 wound up in there. If 10, then I'm curious how 11 wound up in there...

To use 11:
```
rpm --nodeps -e postgresql10{,-contrib,devel,libs,server}
yum install -y postgresql11{,-contrib,devel,libs,server}
```

Otherwise to use 10:
```
rpm --nodeps -e postgresql11-libs
yum install -y postgresql10-libs
```

Then restart Bootstrapper,

```
cd /usr/local/apnscp/resources/playbooks
ansible-playbook bootstrap.yml
```

**Comment by mminglis** [comment]
Opted for 10 through the setup form. I won't even use PostgreSQL, is it used by other services or the panel?

Same issue with bootstrap after following advice, here was output of 10:

```
> rpm --nodeps -e postgresql11-libs
> yum install -y postgresql10-libs
Loaded plugins: etckeeper, fastestmirror, post-transaction-actions, priorities
Loading mirror speeds from cached hostfile
 * epel: fedora.mirror.serversaustralia.com.au
60 packages excluded due to repository priority protections
Package postgresql10-libs-10.7-2PGDG.rhel7.x86_64 already installed and latest version
Nothing to do
```

**Comment by msaladna** [comment]
Mail routing, secondary journal for account metadata, long-term account statistics in apnscp 3.1. Used by Discourse as well if you 1-click it.

Verify that /usr/pgsql-11 is missing then. If not `rm -rf /usr/pgsql-11` (make sure no postgresql11 packages remain). When building PHP it'll enumerate the pgsql-XX directories and take the numerically highest one as the system version.

Then,

```
cd /usr/local/apnscp/resources/playbooks
ansible-playbook bootstrap.yml
```

**Comment by mminglis** [comment]
Cool that fixed it, bootstrapping away. Couple of other things, not urgent...

This:
```
TASK [apnscp/bootstrap : set_fact] **************************************
 [WARNING]: Unable to find '/usr/local/apnscp/config/custom/config.ini' in expected paths (use -vvvvv to see paths)
```

I think is resulting in errors like this:
```
TASK [apnscp/bootstrap : set_fact] ************************************************
fatal: [localhost]: FAILED! => 
  msg: 'An unhandled exception occurred while running the lookup plugin ''ini''. Error was a <class ''ConfigParser.NoSectionError''>, original message: No section: u''ssh'''
...ignoring
```

Also as part of troubleshooting I tried switching to master branch from your post [here](https://forums.apnscp.com/t/apnscp-3-1-development-thread/68/3) and got this:

```
# git checkout --track -b master-31 origin/master-31
fatal: Cannot update paths and switch to branch 'master-31' at the same time.
Did you intend to checkout 'origin/master-31' which can not be resolved as commit?
```

I needed to `git fetch origin` to checkout the 31 branch.

**Comment by mminglis** [comment]
Just an fyi, the errors in this issue are running under master not 31.

**Comment by msaladna** [comment]
Both messages are benign and happen on first run,
```
TASK [apnscp/bootstrap : set_fact] ************************************************
fatal: [localhost]: FAILED! => 
  msg: 'An unhandled exception occurred while running the lookup plugin ''ini''. Error was a <class ''ConfigParser.NoSectionError''>, original message: No section: u''ssh'''
...ignoring
```
```
TASK [apnscp/bootstrap : set_fact] **************************************
 [WARNING]: Unable to find '/usr/local/apnscp/config/custom/config.ini' in expected paths (use -vvvvv to see paths)
```

> I needed to git fetch origin to checkout the 31 branch.

Updated the post. Panel updates will fetch origin automatically, but in case you haven't switched over yet...

PostgreSQL [unified](https://yum.postgresql.org/repopackages.php#pg10) their repo on April 15, which is causing a slew of problems. `python-psycopg2` is pulling down postgresql11-libs during dependency resolution on account of the repo being enabled. I'll have a new release (v3.0.31) released either later tonight or tomorrow to address this.

**Comment by mminglis** [comment]
No worries. Thanks for the help.

**Comment by msaladna** [comment]
Thanks for the bug report. Fixed in [v3.0.31](https://gitlab.com/apisnetworks/apnscp/tags/v3.0.31)
